Ft tmp

cryogenian opened the pull request:
Fixes #1110 

Flexible viz tests fixed. 

@natefaubion  take a look please

natefaubion commented on src/SlamData/Workspace.purs:
```diff
@@ -84,36 +84,42 @@ routeSignal driver =
   workspace
     ∷ UP.DirPath
     → Maybe DeckId
+    → Boolean
```
I'm a little confused by this boolean. The workspace component already does change detection, so can we just push this logic to the workspace component? It's weird to just be passing around a boolean like this.

cryogenian replied:
`putURLVarMaps` send notification, but it must be already set to alter evals behaviour. 
What do you think about sending `urlVarMaps` here instead of boolean?

natefaubion replied:
Yeah, I think passing in the `urlVarMaps` is fine. I'd rather pass the data around and keep the change detection in a single place.

natefaubion commented on test/src/Test/SlamData/Feature/Test/FlexibleVisualization.purs:
```diff
@@ -100,40 +82,58 @@ test =
+                $ SM.singleton did
+                $ SM.singleton "state" "\"NE\""
+          in
+            urlStr ⊕ "/?vars=" ⊕ varsValue
```
This is a pretty useful chunk of code. Maybe we should move this into an `Interact.setUrlVarMap` helper or something?

cryogenian replied:
OK

cryogenian commented:
+ Moved adding varmap to `Interaction`
+ Removed `Boolean` from `Load`. That's a bit tricky. We have guard in route handler, that handle same deck --> there is no need in guards in `Workspace.Component.Load`. As far as I understand using cards cache in `runInitialEval` is for mirroring --> cache should be cleared before loading. Since we actually load stuff in `Load` handler (same routes are filtered) we should reset workspace, decks and caches.

natefaubion commented:
@cryogenian we cannot clear caches when changing decks, that may be as simple as zooming. If we do that that means everything has to run from scratch whenever you zoom in and out. It's ok to clear caches if the workspace is changing though.

cryogenian commented:
Hm, where could I put that `clearCache` then?

natefaubion commented:
I'm not sure there is a clear place (which is why it isn't there, most people don't switch workspaces in the same session, they get to it from the FS). But we should only clear caches when the path is changing. The current setup with `Reset` and `Load` is all wrong, so I would ignore it for now. We should be inserting workspace components with a slot on the path, and the root deck with a slot on the deck id. That's a larger fix though.

cryogenian commented:
Ah, ok if urlvarmaps or path is changed then we can clear cache, right?

natefaubion commented:
@cryogenian It's a little coarse but I guess that's what we should do. Ideally we would only clear caches for cards that depend on url var maps, but that's not information we can figure out right now :)

cryogenian commented:
Like this?

natefaubion commented:
Yeah that works for now.

cryogenian commented:
I have no idea how that's possible, but my branch is currently 11 commits ahead of master and travis passes for it. 

Could you please check tests manually?

cryogenian commented:
Closing because there is #1128
